### ISSUE TITLE
APP_INITIALIZER does not call useFactory in TestBEd

### ISSUE THREAD

**Issue by scottseeker** [angular/angular#16204]
**I'm submitting a ...**  (check one with "x")
```
[X] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
Using APP_INITIALIZER for a service.  Provider syntax uses a useFactory.  This calls the useFactory method in the actual application, but does not call it in the TestBed.
`
TestBed.configureTestingModule({
      imports: [ HttpModule ],
      providers: [
        MockBackend,
          BaseRequestOptions,
          {
            provide: Http,
            deps: [MockBackend, BaseRequestOptions],
            useFactory: (backendInstance: MockBackend, defaultOptions: BaseRequestOptions) => {
              return new Http(backendInstance, defaultOptions);
            }
          },
          Config,
          {
              provide: APP_INITIALIZER,
              useFactory: initConfig,
              deps: [Config],
              multi: true
          }
        ]
    });`

**Expected behavior**

I would expect APP_INITIALIZER to behave in the TestBed the same way it behaves in the applications.  Calls the useFactory method.

**Minimal reproduction of the problem with instructions**

**What is the motivation / use case for changing the behavior?**

Code that runs in application should run in TestBed (with proper TestBed Configuration).

**Please tell us about your environment:**

Windows 10. Visual Code. NPM.

* **Angular version:** 4.0.1

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]

* **Node (for AoT issues):** `node --version` =

**Comment by ridj87** [comment]
@scottseeker  did you get this to work somehow? I'm scratching my head as to why the testbed Config wont load before initalizing the tests.

**Comment by scottseeker** [comment]
@ridj87 No.  I just manually initialized my Config class by calling its load method before the tests.  It says I closed this ticket, but I don't recall closing it or why.

**Comment by vegaed** [comment]
Any way to reopen this?
